### ISSUE TITLE
Lost braces

### ISSUE THREAD

**Issue by bastistician** [r-lib/roxygen2#1578]
Markdown mode in **roxygen2** does not escape braces and this seems to be intended:

https://github.com/r-lib/roxygen2/blob/47e1c76b40b974f4e1c10533c3d41d3889775fac/tests/testthat/test-markdown-link.R#L61-L62

However, many developers don't know that braces need to be escaped in LaTeX-like text in Rd. For example, Rd authors commonly use markup like `\pkg{pkgname}`, but I have seen markdown-type documentation with **roxygen2** using the `{pkgname}` notation to refer to an R package, where these braces are expected to appear in the rendered documentation. This won't happen, unfortunately:

```R
library("roxygen2")
out <- roc_proc_text(rd_roclet(), "
  #' {pkgname}: Playing with Braces
  #'
  #' Plain-text math: integer in {1, 2, ..., n}, (-1)^{n-1}
  #' @md
  null <- function () NULL
")[[1]]

out
#> % Generated by roxygen2: do not edit by hand
#> % Please edit documentation in ./<text>
#> \name{null}
#> \alias{null}
#> \title{{pkgname}: Playing with Braces}
#> \usage{
#> null()
#> }
#> \description{
#> Plain-text math: integer in {1, 2, ..., n}, (-1)^{n-1}
#> }

rdfile <- tempfile()
cat(out$format(), file = rdfile)
tools::Rd2txt(rdfile)
#> pkgname: Playing with Braces
#> 
#> Description:
#> 
#>      Plain-text math: integer in 1, 2, ..., n, (-1)^n-1
#> 
#> Usage:
#> 
#>      null()
#> 

tools::checkRd(rdfile)
#> checkRd: (-3) file2c241344a233:5: Lost braces
#>      5 | \title{{pkgname}: Playing with Braces}
#>        |        ^
#> checkRd: (-3) file2c241344a233:10: Lost braces
#>     10 | Plain-text math: integer in {1, 2, ..., n}, (-1)^{n-1}
#>        |                             ^
#> checkRd: (-3) file2c241344a233:10: Lost braces
#>     10 | Plain-text math: integer in {1, 2, ..., n}, (-1)^{n-1}
#>        |                                                  ^
```

I think it would be nice if **roxygen2** could escape the braces "as necessary" when it generates the Rd code.

**Comment by bastistician** [comment]
@jphill01 This thread is about a bug in **roxygen2**'s Markdown mode. AFAICS, these Rd files in your package weren't even generated by **roxygen2**. You'll find a trivial copy-paste lapsus in the item list if you look at the Rd files. If you are still having problems passing `R CMD check`, you could ask on [R-package-devel](https://stat.ethz.ch/mailman/listinfo/r-package-devel).

(The maintainer's of this repo may wish to mark these comments as off-topic.)

**Comment by hadley** [comment]
I agree that it would be nice to automatically escape these braces, but at this time I don't think it's feasible. The markdown syntax  is a markdown-Rd hybrid parsed by a Frankenstein of commonmark + regular expressions. Unfortunately I don't see a realistic way to improve the situation that either doesn't lead to playing whack-a-mole with new bugs or rewriting the entire parsing system from first principles.

**Comment by bastistician** [comment]
Thanks for taking a look. I can understand the conclusion. Checking with R >= 4.4.0 should note these lost braces anyway, so developers are informed and can switch to specific `\pkg{}` markup etc. or escape the braces as needed.

**Comment by grantmcdermott** [comment]
Just noting that I also got bit by this recently via an `@inherit stats::density details` tag. (The inherited .Rd contains R 4.4+'s `\cite{}\out{<a ...>}` markup, which gets wrapped in extra braces in the generated Rd and triggers `checkRd: Lost braces`.)

Unfortunately, the `\pkg{}` escape workaround obviously doesn't apply here, since the problematic markup is upstream in base R's own Rd files. I'm guessing other folks will run into a similar issue. In my case, it looks like the simplest solution is to drop the `@inherit` tag and just write a short `@details` section pointing to the source instead.

**Comment by bastistician** [comment]
I think these observations are slightly off-topic. Still, a few comments:

- There is currently no citation in the Details section of `?stats::density`. I guess you were inheriting the `bw` argument description from R 4.6.0 with its `\bibcitet{R:Venables+Ripley:2002}`, which gets expanded to a build-time Rd `\Sexpr` that generates an in-text citation that is hyperlinked in HTML help.
- [Inheriting documentation from other packages using roxygen](https://roxygen2.r-lib.org/articles/reuse.html#inheriting-from-other-packages) "is best used sparingly". One undesirable consequence is that all build/install-stage dynamic help fragments from the source become static as on your build machine. In this particular case, the citation target (the reference section) would also be missing.
- I agree the "lost braces" note might be confusing here but it is correct and helpful in the sense that it pointed out a documentation problem.
- Rather than copying the argument documentation from your current R's `?stats::density`, I'd suggest to document `bw` "locally" using a link, for example: "the smoothing bandwidth to be used, see \code{\link[stats]{density}}".

Hope this helps.

PS: I would not call the `\pkg` Rd macro a "workaround"; it is the canonical markup for package names.

**Comment by grantmcdermott** [comment]
Thanks @bastistician, definitely helpful. I'll action on my side. Cheers.
